### PR TITLE
[hlm-707]: Added Update by status in the repositoty, refactored find …

### PR DESCRIPTION
**Pull request by jitendrac-egov**
…by criteria to take data class instead of DTO - jitendra

### CODE REVIEW

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java [hunk 1]
```diff
@@ -60,14 +65,44 @@ void shouldSuccessfullyGetResultsFromSyncRepository()  {
...
+                .id("sync-id")
+                .tenantId("tenant-id")
+                .build();
+
+        Map<String, Object> in = new HashMap<>();
+        in.put("tenantId", "tenant-id");
+        in.put("id", "sync-id");
+        in.put("status", SyncStatus.CREATED.name());
+
+        when(syncLogQueryBuilder.getUpdateSQlBasedOn(any(SyncLogData.class))).thenReturn(GENERATED_UPDATE_QUERY);
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java:95]: Do it similarly in select query

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogQueryBuilder.java [hunk 2]
```diff
@@ -1,35 +1,46 @@
...
             sql.append(" AND fileStoreId=:fileStoreId ");
         }
 
         return sql.toString();
     }
 
+    @Override
+    public String getUpdateSQlBasedOn(SyncLogData syncLogData) {
+        StringBuilder sql = new StringBuilder("UPDATE sync_log SET ");
+        if(syncLogData.getStatus()!=null){
```
Comment (egov-joy) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogQueryBuilder.java:38]: linting / spacing

**Review by jayantp-egov — CHANGES_REQUESTED** [review]

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncLogQueryBuilder.java [hunk 3]
```diff
@@ -1,7 +1,9 @@
 package org.digit.health.sync.repository;
 
-import org.digit.health.sync.web.models.request.SyncLogSearchDto;
+import org.digit.health.sync.web.models.dao.SyncLogData;
 
 public interface SyncLogQueryBuilder {
-    String getSQlBasedOn(SyncLogSearchDto syncLogSearchDto);
+    String getSQlBasedOn(SyncLogData syncLogSearchDto);
```
Comment (egov-joy) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncLogQueryBuilder.java:6]: variable name should also be syncLogData

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java [hunk 4]
```diff
@@ -27,21 +26,33 @@ public DefaultSyncLogRepository(
...
-        in.put("tenantId", syncLogSearchDto.getTenantId());
-        in.put("id", syncLogSearchDto.getSyncId());
-        in.put("status", syncLogSearchDto.getStatus());
-        if( syncLogSearchDto.getReference() != null){
-            in.put("referenceId", syncLogSearchDto.getReference().getId());
-            in.put("referenceIdType", syncLogSearchDto.getReference().getType());
+        in.put("tenantId", syncLogData.getTenantId());
+        in.put("id", syncLogData.getId());
+        in.put("status", syncLogData.getStatus());
+        if( syncLogData.getReferenceId() != null){
```
Comment (egov-joy) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java:34]: linting / spacing

**Review by egov-joy — CHANGES_REQUESTED** [review]

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java (hunk 4 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/DefaultSyncLogRepository.java:34]: done

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncLogQueryBuilder.java (hunk 3 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncLogQueryBuilder.java:6]: done

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncLogUpdateDto.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package org.digit.health.sync.web.models.request;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
```
[the remark this answers is not in the document]
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/web/models/request/SyncLogUpdateDto.java:4]: done

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java [hunk 6]
```diff
@@ -42,16 +43,20 @@ void setUp() {
         defaultSyncLogRepository = new DefaultSyncLogRepository(jdbcTemplate, syncLogQueryBuilder);
     }
 
+
     @Test
     @DisplayName("should successfully get results from sync repository")
     void shouldSuccessfullyGetResultsFromSyncRepository()  {
-        SyncLogSearchRequest syncLogSearchRequest = SyncSearchRequestTestBuilder.builder().withReferenceId().build();
+        SyncLogSearchDto syncLogSearchDto = SyncLogSearchDto.builder().reference(ReferenceId.builder()
```
[the remark this answers is not in the document]
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java:50]: done

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java [hunk 7]
```diff
@@ -42,16 +43,20 @@ void setUp() {
...
+                .type("campaign")
+                .id("id")
+                .build()).build();
         List<SyncLogData> searchedData = new ArrayList<>();
         searchedData.add(SyncLogData.builder().build());
-        SyncLogSearchDto syncLogSearchDto = SyncLogSearchMapper.INSTANCE.toDTO(syncLogSearchRequest);
+        SyncLogData syncLogData = SyncLogSearchMapper.INSTANCE.toData(syncLogSearchDto);
```
[the remark this answers is not in the document]
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java:56]: done

File: digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java (hunk 1 above)
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/test/java/org/digit/health/sync/repository/DefaultSyncLogRepositoryTest.java:95]: done

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncLogQueryBuilder.java [hunk 8]
```diff
@@ -1,7 +1,9 @@
 package org.digit.health.sync.repository;
 
-import org.digit.health.sync.web.models.request.SyncLogSearchDto;
+import org.digit.health.sync.web.models.dao.SyncLogData;
 
 public interface SyncLogQueryBuilder {
-    String getSQlBasedOn(SyncLogSearchDto syncLogSearchDto);
```
[the remark this answers is not in the document]
Comment (jitendrac-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/SyncLogQueryBuilder.java:6]: done

**Review by jayantp-egov — APPROVED** [review]

**Review by egov-joy — APPROVED** [review]

**Pull request merged**